tests/arch/arm/arm_irq_advanced_features/src/arm_dynamic_direct_interrupts.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <arch/cpu.h>
+#include <arch/arm/cortex_m/cmsis.h>
+
+
+void arm_direct_isr_handler_0(void *args)
+{
+  ARG_UNUSED(args);
+
+	printk("test\n");
+}
+
+void test_arm_dynamic_direct_interrupts(void)
+{
+  /* Place the dynamic interrupt dispatcher in the ROM ISR table. */
+	IRQ_DIRECT_CONNECT(TIMER0_IRQn, 0,
+		z_arm_irq_direct_dynamic_dispatch_no_reschedule, 0);
```

Comment:
Perhaps we could create a simpler way to declare this ISR?

Now the application has to reference the ISR direct dynamic wrapper. Instead we could just have something that declared TIMER0_IRQn has an interrupt that needs to be connected dynamically.
That is what I wanted to solve with this:
https://github.com/zephyrproject-rtos/zephyr/pull/20387/files#diff-647d5b552bb42878dc0504d70ee3ac50R56

Maybe the tests have to be written this way, but the application should have something that doesn't reference reference zephyr internal functions.

Reply:
It does not reference internal function; it references an ARM-only API.
I am OK with removing the z_ prefix, to indicate _better_ that this is not internal function. But, somehow, this is an architecture-specific API, so maybe call it simply: `arm_irq_direct_dynamix_dispatch_no_reschedule`?

@carlescufi what is the convention for ARCH-specific APIs exposed, though, publicly?

Reply:
>@carlescufi what is the convention for ARCH-specific APIs exposed, though, publicly?

Not an expert here, @andrewboie is the person to ask, but from what I can tell functions in `include/arch/` seem to be prefixed with `arch_`

Reply:
Alright